### PR TITLE
feat(as): add a datasource to get list of AS instances hanging information

### PR DESCRIPTION
**Pull request by ruwenqiang123**
**What this PR does / why we need it**:

Add a new datasource to get list of AS instances hanging information.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1.support a new datasource to get the list of AS instances hanging information.
2.support related document and acceptance test.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
$ make testacc TEST="./huaweicloud/services/acceptance/as" TESTARGS="-run TestAccDataSourceHookInstances_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/as -v -run TestAccDataSourceHookInstances_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceHookInstances_basic
=== PAUSE TestAccDataSourceHookInstances_basic
=== CONT  TestAccDataSourceHookInstances_basic
--- PASS: TestAccDataSourceHookInstances_basic (35.74s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/as        35.818s
```

### CODE REVIEW

File: docs/data-sources/as_hook_instances.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+page_title: "HuaweiCloud: huaweicloud_as_hook_instances"
+description: |-
+  Use this data source to get a list of AS instances hanging information.
+---
+
+# huaweicloud_as_hook_instances
+
+Use this data source to get a list of AS instances hanging information.
+
+-> **NOTE:** After a lifecycle hook is added, when an AS group performs a scaling action, the lifecycle hook suspends
```
Comment (deer-hang) [inline docs/data-sources/as_hook_instances.md]: remove this line.

File: docs/data-sources/as_hook_instances.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  the target instance and sets it to be in wating state.
+  <br/>You can query the instance suspension based on search criteria.
+
+## Example Usage
+
+```hcl
+variable "scaling_group_id" {}
+variable "instance_id" {}
+
+data "huaweicloud_as_planned_tasks" "test" {
```
Comment (deer-hang) [inline docs/data-sources/as_hook_instances.md]:
```suggestion
data "huaweicloud_as_hook_instances" "test" {

```

File: huaweicloud/services/acceptance/as/data_source_huaweicloud_as_hook_instances_test.go [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+)
+
+func TestAccDataSourceHookInstances_basic(t *testing.T) {
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/as/data_source_huaweicloud_as_hook_instances_test.go:13]: Add a note: 
// Before running this test case, you need to prepare an AS group ID with hook instances.

File: huaweicloud/services/acceptance/as/data_source_huaweicloud_as_hook_instances_test.go [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+				),
+			},
+		},
+	})
+}
+
+func testDataSourceHookInstances_basic() string {
+	return fmt.Sprintf(`
+data "huaweicloud_as_hook_instances" "test" {
+  scaling_group_id      = "%[1]s"
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/as/data_source_huaweicloud_as_hook_instances_test.go]: too many spaces.

File: huaweicloud/services/acceptance/acceptance.go [hunk 5]
```diff
@@ -28,6 +28,8 @@ var (
 	HW_PROJECT_ID_2       = os.Getenv("HW_PROJECT_ID_2")
 	HW_PROJECT_ID_3       = os.Getenv("HW_PROJECT_ID_3")
 
+	HW_AS_GROUP_ID = os.Getenv("HW_AS_GROUP_ID")
```
Comment (jinyangyang222) [inline huaweicloud/services/acceptance/acceptance.go:31]: Using existing environment variable, HW_AS_SCALING_GROUP_ID

**Pull request merged**